plugin/plugin.go

```diff
@@ -162,6 +137,108 @@ func (e *Ratelimit) Type() rules.ActionType {
 	return rules.ActionTypeNondisruptive
 }
 
+func (e *Ratelimit) parseConfig(config string) error {
+	// acceptable keys
+	var err error
+
+	tokens := strings.Split(config, "&")
+
+	requiredValues := map[string]bool{
+		"zone":   false,
+		"events": false,
+		"window": false,
+	}
+
+	for _, token := range tokens {
+		pair := strings.Split(token, "=")
+		if len(pair) != 2 {
+			return fmt.Errorf("invalid usage of = for %v", pair)
+		}
+
+		key := pair[0]
+		value := pair[1]
+
+		switch key {
+		case "zone":
+			if e.ZoneMacro, err = macro.NewMacro(value); err != nil {
+				return fmt.Errorf("invalid macro name")
+			}
+			requiredValues[key] = true
+		case "events":
+			if e.MaxEvents, err = strconv.ParseInt(value, 10, 64); err != nil {
+				return fmt.Errorf("invalid integer value for events")
+			}
+			requiredValues[key] = true
+		case "window":
+			if e.Window, err = strconv.ParseInt(value, 10, 64); err != nil {
+				return fmt.Errorf("invalid integer value for window")
+			}
+			if e.Window == 0 {
+				return fmt.Errorf("value 0 is not allowed for key 'window'")
+			}
+			requiredValues[key] = true
+		case "interval":
+			if e.SweepInterval, err = strconv.ParseInt(value, 10, 64); err != nil {
```

Comment:
I believe it is better to store this as `time.Duration`

Reply:
changing right away

Reply:
done